src/main/java/bgu/spl/net/impl/stomp/LibraryProtocol.java

```diff
@@ -13,59 +13,168 @@
     private Client _client;
     private int _connectionId;
     private Connections<Frame> _connections;
+    private Integer _messagesSent;
 
     @Override
     public void start(int connectionId, Connections<Frame> connections) {
         _connectionId = connectionId;
         _connections = connections;
+        _messagesSent = 0;
     }
 
     @Override
     public void process(Frame msg) {
         switch (msg.getCommand()) {
             case CONNECT:
-                _connections.send(_connectionId, handleConnect(msg));
+                handleConnect(msg);
                 break;
             case SEND:
                 handleSend(msg);
                 break;
-            // TODO: Add the rest of the handling. For all of the commands
+            case SUBSCRIBE:
+                handleSubscribe(msg);
+                break;
+            case UNSUBSCRIBE:
+                handleUnsubscribe(msg);
+                break;
+            case DISCONNECT:
+                handleDisconnect(msg);
+                break;
         }
     }
 
+    private void handleDisconnect(Frame msg) {
+        sendReceipt(msg);
+        _connections.disconnect(_connectionId);
+        _client.connected = false;
+        _shouldTerminate = true;
```

Comment:
and why is that?

Reply:
What are you asking about? this is how you disconnect a user.
Are you asking about the `_shouldTerminate=true`? because the protocol is per client and should terminate once the user does

Reply:
oh okay. so you don't kill the server with this _shouldTerminate